File: pos_credit_limit_alert_20072025/models/pos_credit_limit.py
from odoo import api, models

class PosCreditLimit(models.Model):
    _inherit = "res.partner"

    @api.model
    def check_partner_credit(self, partner_id):
        partner = self.browse(partner_id)
        if not partner:
            return {'allowed': True, 'message': ''}

        credit_limit = partner.credit_limit
        debt = partner.credit or 0.0

        if debt > credit_limit:
            message = (
                f"El cliente {partner.name} tiene un límite de crédito de "
                f"{credit_limit} y una deuda actual de {debt}. No puede continuar."
            )
            return {'allowed': False, 'message': message}

        return {'allowed': True, 'message': ''}

File: pos_credit_limit_alert_20072025/static/src/js/pos_credit_limit.js
odoo.define('pos_credit_limit_alert.pos_credit_limit', function (require) {
    "use strict";

    const PosModel = require('point_of_sale.models');
    const Registries = require('point_of_sale.Registries');
    const { Gui } = require('point_of_sale.Gui');

    const PosCreditLimit = (PosModel) => class extends PosModel {
        async _flush_orders() {
            const partner = this.get_client();

            if (partner) {
                const res = await this.rpc({
                    model: 'res.partner',
                    method: 'check_partner_credit',
                    args: [partner.id],
                });

                if (!res.allowed) {
                    await Gui.showPopup('ErrorPopup', {
                        title: 'Límite de Crédito Excedido',
                        body: res.message,
                    });
                    return false;
                }
            }

            return super._flush_orders();
        }
    };

    Registries.Model.extend(PosModel, PosCreditLimit);

    return PosModel;
});
